x-pack/plugins/observability/server/services/slo/find_slo_groups.ts

```diff
@@ -0,0 +1,205 @@
+  };
+}
+
+interface TopHitsResult {
+  _source: {
+    sliValue: number;
+    status: Status;
+  };
+}
+interface Aggregation {
+  doc_count: number;
+  key: string;
+  worst: {
+    hits: {
+      hits: TopHitsResult[];
+    };
+  };
+  violated: {
+    doc_count: number;
+  };
+  healthy: {
+    doc_count: number;
+  };
+  degrading: {
+    doc_count: number;
+  };
+  noData: {
+    doc_count: number;
+  };
+}
+
+interface GroupAggregationsResponse {
+  groupBy: {
+    buckets: Aggregation[];
+  };
+  distinct_items: {
+    value: number;
+  };
+}
+
+export class FindSLOGroups {
+  constructor(
+    private esClient: ElasticsearchClient,
+    private logger: Logger,
+    private spaceId: string
+  ) {}
+  public async execute(params: FindSLOGroupsParams): Promise<FindSLOGroupsResponse> {
+    const pagination = toPagination(params);
+    const groupBy = params.groupBy;
+    const kqlQuery = params.kqlQuery ?? '';
+    const filters = params.filters ?? '';
+    let parsedFilters: any = {};
+
+    try {
+      parsedFilters = JSON.parse(filters);
+    } catch (e) {
+      this.logger.error(`Failed to parse filters: ${e.message}`);
+    }
+    const response = await this.esClient.search<unknown, GroupAggregationsResponse>({
+      index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { spaceId: this.spaceId } },
+            getElastichsearchQueryOrThrow(kqlQuery),
+            ...(parsedFilters.filter ?? []),
+          ],
+        },
+      },
+      body: {
+        aggs: {
+          groupBy: {
+            terms: {
+              field: groupBy,
+              size: 10000,
+            },
+            aggs: {
+              worst: {
+                top_hits: {
+                  sort: {
+                    errorBudgetRemaining: {
+                      order: 'asc',
+                    },
+                  },
+                  _source: {
+                    includes: ['sliValue', 'status'],
+                  },
+                  size: 1,
+                },
+              },
+              violated: {
+                filter: {
+                  term: {
+                    status: 'VIOLATED',
+                  },
+                },
+              },
+              healthy: {
+                filter: {
+                  term: {
+                    status: 'HEALTHY',
+                  },
+                },
+              },
+              degrading: {
+                filter: {
+                  term: {
+                    status: 'DEGRADING',
+                  },
+                },
+              },
+              noData: {
+                filter: {
+                  term: {
+                    status: 'NO_DATA',
+                  },
+                },
+              },
+              bucket_sort: {
+                bucket_sort: {
+                  sort: [
+                    {
+                      _key: {
```

Comment:
@kdelemme We discussed about this one, and if I recall correctly you suggested to use `groupBy` instead of _key? I used `_key` cause I found it [here](https://stackoverflow.com/questions/64386038/elasticsearch-bucket-sort-aggregation-sorting-on-alphabetical-field). If you think we should change this, feel free to change